src/price_providers/endpoint_auction_pricing.py

```diff
@@ -43,8 +43,11 @@ def get_price(self, price_params: dict) -> float | None:
                 return price_in_eth
 
             except requests.exceptions.HTTPError as err:
-                if err.response.status_code == 404:
-                    pass
+                # Continue to check if tx present on barn.
+                if err.response.status_code == 404 and environment == "prod":
```

Comment:
Does this assume that "prod" is checked first? This seems a somewhat risky assumption, and I don't think we should rely on the order in which the `endpoint_urls` entries are parsed.

i would rather add a counter before the for-loop and just count if there are any unchecked items